resource.go

```diff
@@ -503,13 +560,25 @@ func toProto(resource Resource) *pb.Resource {
 	return b
 }
 
+func timestampFromProto(ts *tspb.Timestamp) time.Time {
```

Comment:
Let's return `time.Time{}` when `ts == nil`. This will clearly distinguish missing timestamps, for example from old manifests that don't implement timestamping, or ones generated on OSs that don't support every field. Otherwise we would return the unix epoch here.